src/vs/workbench/contrib/terminal/browser/terminalEditorService.ts

```diff
@@ -54,16 +54,13 @@ export class TerminalEditorService extends Disposable implements ITerminalEditor
 	}
 
 	private _setActiveInstance(instance: ITerminalInstance | undefined): void {
-		const oldActiveInstance = this.activeInstance;
 		if (instance === undefined) {
 			this._activeInstanceIndex = -1;
 		} else {
 			this._activeInstanceIndex = this.instances.findIndex(e => e === instance);
 		}
 		const newActiveInstance = this.activeInstance;
-		if (oldActiveInstance !== newActiveInstance) {
```

Comment:
@Tyriar this was causing some weird issues. I tried to do something similar in the terminal service refactor and it caused an obscure bug.